test/models/record_test.rb

```diff
@@ -27,4 +27,33 @@ def test_record_should_be_able_to_find_calculation_results
     r = Record.find('51703a883054cf84390000d4')
     assert_equal 2, r.calculation_results.count, 'record should have 2 calculated results'
   end
+
+  def assert_record_demographics_equal(r1, r2, assert_true = true)
```

Comment:
The `assert_true` argument and the assert message seem a little awkward to me. I think a slightly cleaner way might be to make it `record_demographics_equal(r1, r2)` and return a bool. So then calling it will be like:
`assert(record_demographics_equal(r1, r2), message)`
or
`assert(!record_demographics_equal(r1, r2), message)`
You could use that to make the messages more specific to which comparison you're making, e.g. `'record demographics should be equal after duplication'` for the one on line 56